dataactvalidator/scripts/load_cfda_data.py

```diff
@@ -13,86 +15,138 @@
-             "regulations__guidelines__and_literature_(140)": "regulations_guidelines_and_literature",
-             "regional_or__local_office_(151)": "regional_or_local_office",
-             "headquarters_office_(152)": "headquarters_office",
-             "website_address_(153)": "website_address",
-             "related_programs_(160)": "related_programs",
-             "examples_of_funded_projects_(170)": "examples_of_funded_projects",
-             "criteria_for_selecting_proposals_(180)": "criteria_for_selecting_proposals",
-             "url": "url",
-             "recovery": "recovery",
-             "omb_agency_code": "omb_agency_code",
-             "omb_bureau_code": "omb_bureau_code",
-             "published_date": "published_date",
-             "archived_date": "archived_date"},
+            DATA_CLEANING_MAP,
             {}
         )
-        data["published_date"] = format_date(data["published_date"])
-        data["archived_date"] = format_date(data["archived_date"])
+        import_data["published_date"] = format_date(import_data["published_date"])
+        import_data["archived_date"] = format_date(import_data["archived_date"])
+        import_dataframe = import_data.copy(deep=True)
+        # To do the comparison, first we need to mock the pk column that postgres creates. We'll set it universally to 1
+        import_dataframe = import_dataframe.assign(cfda_program_id=1, created_at=now, updated_at=now)
 
-        # insert to db
         table_name = model.__table__.name
-        num = insert_dataframe(data, table_name, sess.connection())
-        sess.commit()
+        current_data = pd.read_sql_table(table_name, sess.connection(), coerce_float=False)
+        # Now we need to overwrite the db's audit dates in the created dataframe, and
+        # ALSO set all the  pks to 1, so they match
+        current_data = current_data.assign(cfda_program_id=1, created_at=now, updated_at=now)
+        # pandas comparison requires everything to be in the same order
+        current_data.sort_values('program_number', inplace=True)
+        import_dataframe.sort_values('program_number', inplace=True)
 
-    logger.info('{} records inserted to {}'.format(num, table_name))
+        # columns too
+        cols = import_dataframe.columns.tolist()
+        cols.sort()
+        import_dataframe = import_dataframe[cols]
+
+        cols = current_data.columns.tolist()
+        cols.sort()
+        current_data = current_data[cols]
+
+        # need to reset the indexes now that we've done all this sorting, so that THEY match
+        import_dataframe.reset_index(drop=True, inplace=True)
+        current_data.reset_index(drop=True, inplace=True)
+        # My favorite part: When pandas pulls the data out of postgres, the program_number column
+        # is a Decimal. However, in adding it to the dataframe, this column loses precision, for
+        # reasons I could not uncover and cannot fathom. So for example, a program number of
```

Comment:
There is probably a good reason for this happening... I would avoid "I couldn't figure out how to do this" explanation, and just explain what is.